src/app/(tabs)/index.tsx

```diff
@@ -1,69 +1,336 @@
+  const [allProperties, setAllProperties] = useState<Property[]>([]);
+  const [filteredProperties, setFilteredProperties] = useState<Property[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [filters, setFilters] = useState<FilterState>(globalFilters);
+
+  // Filter function
+  useEffect(() => {
+    globalApplyFilter = (newFilters: FilterState): void => {
+      console.log("Applying filters in index:", newFilters);
+      setFilters(newFilters);
+      globalFilters = newFilters;
+    };
+
+    return (): void => {
+      globalApplyFilter = null;
+    };
+  }, []);
+
+  // Using client sided filtering as firestore doesn't support complex "OR" queries
+  // Eg, both "minimum bedrooms" and "minimum bathrooms" at the same time
+  const applyFilters = (
+    properties: Property[],
+    filters: FilterState,
+  ): Property[] => {
+    return properties.filter((property) => {
+      // Property type filter - if empty, show all types
+      if (filters.type.length > 0) {
+        const propertyType = property.type || "rental";
+        if (!filters.type.includes(propertyType)) {
+          return false;
+        }
+      }
+
+      // Price filter
+      if (filters.minPrice !== "" || filters.maxPrice !== "") {
+        const minPrice = filters.minPrice ? parseFloat(filters.minPrice) : 0;
+        const maxPrice = filters.maxPrice
+          ? parseFloat(filters.maxPrice)
+          : Infinity;
+
+        if (!isNaN(minPrice) && !isNaN(maxPrice)) {
+          if (property.price < minPrice || property.price > maxPrice) {
+            return false;
+          }
+        }
+      }
+
+      // Bedrooms filter - single selection (null means no filter)
+      if (filters.bedrooms !== null) {
+        const propertyBedrooms = property.bedrooms || 0;
+        if (propertyBedrooms < filters.bedrooms) {
+          return false;
+        }
+      }
+
+      // Bathrooms filter - single selection (null means no filter)
+      if (filters.bathrooms !== null) {
+        const propertyBathrooms = property.bathrooms || 0;
+        if (propertyBathrooms < filters.bathrooms) {
+          return false;
+        }
+      }
+
+      // Contract length filter
+      if (filters.minContract && filters.minContract !== "") {
+        const minContract = parseInt(filters.minContract);
+        if (!isNaN(minContract)) {
+          if (!property.contract || property.contract > minContract) {
+            return false;
+          }
+        }
+      }
+
+      return true;
+    });
+  };
+
+  // Apply filters whenever filters change
+  useEffect(() => {
+    const filtered = applyFilters(allProperties, filters);
+    console.log(
+      `Filtered ${filtered.length} properties from ${allProperties.length} total`,
+    );
+    console.log("Current filters:", filters);
+    setFilteredProperties(filtered);
+  }, [allProperties, filters]);
+
+  // Fetch properties from Firebase using v9+ API
+  useEffect(() => {
+    const db = getFirestore();
```

Comment:
```suggestion
  const uid = getAuth().currentUser?.uid;

  // Fetch properties from Firebase using v9+ API
  useEffect(() => {
    if (!uid) {
      return;
    }

    const db = getFirestore();
```
Check whether user has logged in.